### ISSUE TITLE
setupSnapshot() Causing AWS Failure

### ISSUE THREAD

**Issue by cdunkel** [fastlane/fastlane#4366]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

This is an issue I'm running into while setting up XCode 7 UI Tests to support the latest version of Snapshot.  I have been following the [UI Tests Quick Start guide](https://github.com/fastlane/fastlane/tree/master/snapshot#ui-tests), but have run into a problem running my tests with Snapshot from XCode 7.

**Complete Contents of My Test Swift File:**

```
import XCTest

class CFPlusUITests_Snapshot: XCTestCase {
        
    override func setUp() {
        super.setUp()
        
        // Put setup code here. This method is called before the invocation of each test method in the class.
        
        // In UI tests it is usually best to stop immediately when a failure occurs.
        continueAfterFailure = false
        // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
        let app = XCUIApplication()
        setupSnapshot(app)
        app.launch()

        // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
    }
    
    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }
    
    func testExample() {
        
        let app = XCUIApplication()
        
        sleep(10)
        
        // Just open the app, select the first fair, and then wait for the dashboard to appear
        let firstChild = app.tables.childrenMatchingType(.Cell).elementBoundByIndex(0);
        if (firstChild.exists) {
            NSLog("First child does exist!");
            firstChild.tap()
        } else {
            NSLog("First child does not exist.");
        }
        
        sleep(5)
    }   
}
```

Running these tests from XCode using `Command-U` successfully launches the app, but I receive the following in-app error when attempting to connect to our AWS S3 bucket:

![simulator screen shot apr 1 2016 11 36 33 am](https://cloud.githubusercontent.com/assets/685932/14821152/a3c9ab70-0b97-11e6-918d-ed1f0746a305.png)

If, however, I simply comment out the `setupSnapshot(app)` line from the above code the app runs successfully.

I am very new to using UITests, so I am open to any advice for the best way to debug this issue.  So far I have not been able to get any additional information about why this might be occurring.

**Comment by TKBurner** [comment]
Hi @cdunkel. Thanks for reaching out about this. Are you using any type of special transport security with your S3 bucket or just direct SSL? The reason I ask is because this error suggests that there is some sort of request signature that's being altered from it's expected value. Thanks!

**Comment by cdunkel** [comment]
@TKBurner I'm not doing anything special.  All of the transport security is wrapped up in the AWS iOS SDK that I'm using, which has fairly terrible documentation, so I'm not sure exactly what they're using under the covers.  I'm simply providing an access key and a secret key.

**Comment by TKBurner** [comment]
@cdunkel Thanks for following up. I'm not too familiar with AWS iOS SDK either. This seems like it's something related to your integration. Check out this page for some info on that specific error:
http://docs.aws.amazon.com/general/latest/gr/signature-v4-troubleshooting.html and this issue in their github repo https://github.com/aws/aws-sdk-js/issues/86

**Comment by cdunkel** [comment]
@TKBurner Thanks for the response!  I had actually come across the doc you referenced previously, but not the issue.  That could be helpful!  I will spend some time today or tomorrow and see if I can get some more logs from my app that might shed some light onto why the `setupSnapshot(app)` call might be causing issues.

**Comment by TKBurner** [comment]
@cdunkel Awesome! Let me know how it goes 👍

**Comment by cdunkel** [comment]
**Update:** My captured app logs didn't really give me any more information on this.  I have posted an issue to the AWS iOS SDK repo, too, to get some advice from that side of things.

https://github.com/aws/aws-sdk-ios/issues/381

**Comment by cdunkel** [comment]
I figured it out!

Part of what the `setupSnapshot(app)` function does is to set the *language* for the app.  It does this using the **language.txt** file located in `~/Library/Caches/tools.fastlane`.

However, if you have not run `snapshot` from the console yet, then the **language.txt** file does not exist and, rather than throwing an error, it sets the simulator device language to "".  This doesn't seem to affect the ability for the app to run in the simulator, but does cause requests to AWS S3 to fail.

So the "solution" was to run `snapshot` in the console first, but it seems like the `setupSnapshot(app)` function should really fail out if that language file is not found, or at least fail in a way that does not adversely affect the operation of the app.  

My Swift isn't great, but it appears that `deviceLanguage = try NSString(contentsOfFile: path, encoding: NSUTF8StringEncoding).stringByTrimmingCharactersInSet(trimCharacterSet) as String` is happy to return `nil` if the file at `path` doesn't exist.  What should probably happen is that returned var should be checked before setting the language.

I can create a PR for that change, if desired.

**Comment by TKBurner** [comment]
@cdunkel Thanks for following up. This is expected behavior. Did you run snapshot init in this folder before you ran the lane? That seems to be where things broke. 

Thanks!

**Comment by cdunkel** [comment]
@TKBurner I'm actually using Snapshot independently of a lane, since most of my iOS build structure was implemented before Fastlane combined all the tools.  I did run `snapshot init` in my iOS project folder before attempting to link Snapshot up with my tests, but the **language.txt** file I referenced only seems to be created on actually running `snapshot` from the console (or through a lane, I assume).  So my real mistake was attempting to run by UITest target through XCode before running `snapshot` through the console.

It still seems odd to me that if the **language.txt** file doesn't exist - for whatever reason - the `setLanguage()` function in SnapshotHelper.swift doesn't hit the "Couldn't detect/set language..." block.
